attenuation.py

```diff
@@ -0,0 +1,513 @@
+        """
+        Calculate V-band normalised optical depth.
+
+        Args:
+            lam (float/array-like, float)
+                An array of wavelengths or a single wavlength at which to
+                calculate optical depths (in AA, global unit).
+
+        Returns:
+            float/array-like, float
+                The optical depth.
+        """
+        if isinstance(lam, (unyt_quantity, unyt_array)):
+            _lam = lam.to('Angstrom').v
+        else:
+            _lam = lam
+        return self.emodel((_lam*Angstrom).to_astropy())
+
+    def get_transmission(self, tau_v, lam):
+        """
+        Returns the transmitted flux/luminosity fraction based on an optical
+        depth at a range of wavelengths.
+
+        Args:
+            tau_v (float/array-like, float)
+                Optical depth in the V-band. Can either be a single float or
+                array.
+
+            lam (array-like, float)
+                The wavelengths (with units) at which to calculate transmission.
+
+        Returns:
+            array-like
+                The transmission at each wavelength. Either (lam.size,) in shape
+                for singular tau_v values or (tau_v.size, lam.size) tau_v
+                is an array.
+        """
+        if isinstance(lam, (unyt_quantity, unyt_array)):
+            _lam = lam.to('Angstrom').v
+        else:
+            _lam = lam
+        return self.emodel.extinguish(x=(_lam*Angstrom).to_astropy(), Av=1.086 * tau_v)
+
+
+
+def Li_08(lam, c1, c2, c3, c4):
+        """
+        Drude-like parametric expression for the attenuation curve from Li+08
+        
+        Args:
+            
+            lam (array-like, float)
+                The wavelengths (with unit == micron) at which to calculate transmission.
+        
+            c1, c2, c3, c4 (float)
+                Dimensionless parameters describing the FUV rise, optical/NIR extinction and UV bump strength.
+                Values for the Calzetti, SMC, MW and LMC attenuation curve (Li+08):
+                    Calzetti SMC    MW      LMC
+                c1  44.9     38.7   14.4    4.47
+                c2  7.56     3.83   6.52    2.39
+                c3  61.2     6.34   2.04    –0.988
+                c4  0        0      0.0519  0.0221
+        
+        
+        Returns:
+            array-like
+                tau/tau_v at each input wavelength (lam)
+        
+        """
+        
+        # Wavelength in microns
+        if isinstance(lam, (unyt_quantity, unyt_array)):
+            lam_micron = lam.to('um').v
+        else:
+            lam_micron = lam/1e4
+            
+        
+        
+        #Attenuation curve normalized to Av
+        A_lam_v = c1 / ((lam_micron/0.08)**c2 + (lam_micron/0.08)**-c2 + c3)  +   (233. * (1 - c1/(6.88**c2 + 0.145**c2 +c3) - c4/4.6)) / ((lam_micron/0.046)**2. + (lam_micron/0.046)**-2. + 90.)  +  c4/ ((lam_micron/0.2175)**2. + (lam_micron/0.2175)**-2. - 1.95)
```

Comment:
And here

Reply:
Could this line be broken up for readability? (And keeping to 79 characters per line).